src/ResourceManagement/Locks/ManagementLocksImpl.cs

```diff
@@ -0,0 +1,308 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+namespace Microsoft.Azure.Management.Locks.Fluent
+{
+    using System.Threading;
+    using System.Threading.Tasks;
+    using System.Collections.Generic;
+    using Microsoft.Azure.Management.Locks.Fluent.Models;
+    using Microsoft.Azure.Management.ResourceManager.Fluent.Core;
+    using System.Text;
+    using System;
+
+    /// <summary>
+    /// Implementation for ManagementLocks.
+    /// </summary>
+    ///GENTHASH:Y29tLm1pY3Jvc29mdC5henVyZS5tYW5hZ2VtZW50LmxvY2tzLmltcGxlbWVudGF0aW9uLk1hbmFnZW1lbnRMb2Nrc0ltcGw=
+    internal partial class ManagementLocksImpl :
+        CreatableResources<IManagementLock, ManagementLockImpl, Models.ManagementLockObjectInner>,
+        IManagementLocks
+    {
+        private IAuthorizationManager manager;
+
+                ///GENMHASH:836797384EAD11F4F592C5C904884C2A:D544D70B59C64F2C67EC02DE16CBAEF6
+        internal ManagementLocksImpl(IAuthorizationManager manager)
+        {
+            this.manager = manager;
+        }
+
+        IManagementLocksOperations IHasInner<IManagementLocksOperations>.Inner
+        {
+            get
+            {
+                return Inner();
+            }
+        }
+
+        ///GENMHASH:B6961E0C7CB3A9659DE0E1489F44A936:168EFDB95EECDB98D4BDFCCA32101AC1
+        public IAuthorizationManager Manager()
+        {
+            return manager;
+        }
+
+        #region Get
+
+        ///GENMHASH:7D93B4EC99C64989F97B3D17F88C3F2C:45E96D80A053DE48AD3DD26FC5CD83FF
+        public IManagementLock GetByResourceGroup(string resourceGroupName, string name)
+        {
+            return Extensions.Synchronize(() => GetByResourceGroupAsync(resourceGroupName, name));
+        }
+
+        ///GENMHASH:5002116800CBAC02BBC1B4BF62BC4942:E5DB8B81D366BCF6A1B75F4B1D85B1F1
+        public IManagementLock GetById(string id)
+        {
+            return Extensions.Synchronize(() => GetByIdAsync(id));
+        }
+
+        ///GENMHASH:E776888E46F8A3FC56D24DF4A74E5B74:FC67528805BD588653B69DE503549F4C
+        public async Task<IManagementLock> GetByIdAsync(string id, CancellationToken cancellationToken = default(CancellationToken))
+        {
+            string resourceId = ResourceIdFromLockId(id);
+            string lockName = ResourceUtils.NameFromResourceId(id);
+            IManagementLock l;
+            ManagementLockObjectInner inner = null;
+            if (resourceId == null || lockName == null)
+            {
+                l = null;
+            }
+            else if (null == (inner = await Inner().GetByScopeAsync(resourceId, lockName)))
+            {
+                l = null;
+            }
+            else
+            {
+                l = WrapModel(inner);
+            }
```

Comment:
nit: Looks like this can be simplified in a single if statement 
```
IManagementLock l=null;
if( resourceId != null && lockName != null) 
{
    l = WrapModel(await Inner().GetByScopeAsync(resourceId, lockName));
}
```
later in the code in the WrapModel you do check if ManagementLockObjectInner passed in is null and just return null.

Reply:
good point